src/submit.rs

```diff
@@ -476,4 +477,44 @@ impl<'a> Submitter<'a> {
         )
         .map(drop)
     }
+
+    /// Performs a synchronous cancellation request, similiar to [AsyncCancel](crate::opcode::AsyncCancel),
+    /// except that it completes synchronously.
+    ///
+    /// [`AsyncCancelFlags`](types::AsyncCancelFlags) may be supplied to indicate the strategy for
+    /// matching operations to cancel.
+    ///
+    /// Available since 6.0.
+    pub fn register_sync_cancel(
+        &self,
+        user_data: u64,
+        fd: RawFd,
+        timeout: Option<Duration>,
```

Comment:
I had a few questions with the public API here:

1. Do we want to take a `Duration` or a `Timespec`? 
2. Users can cancel based on either a "regular" FD or a fixed FD via the `AsyncCancelFlags`. Should we remove the distinction in the flags and automatically pick the correct flag based on the `fd` argument? I think that would mean that we'd take `impl sealed::UseFixed` instead of a raw fd.

Reply:
1. I think using `Timespec` has more consistency.
2. for consistency reasons, `UseFixed` would be good.
3. I think `None` for infinite is good because `Timespec::new()` can easily express zero.